examples/instance/terraform/plugin.go

```diff
@@ -343,6 +371,16 @@ func (p *plugin) Provision(spec instance.Spec) (*instance.ID, error) {
 
 	id := instance.ID(name)
 
+	storageType, storageName, storageProps, err := FindStorage(&tf)
```

Comment:
I wonder if this could be handled similar to the `optionalProcessHostname()` function in line 401.  Say, let's have a `optionalProcessStorageName()` method that gets triggered when there's a `vmType` vm that matches `VMSoftLayer`.   This way, we can be very clear about how this logic applies only to Softlayer resources.   Right now,  anything that's in the `StorageTypes` slice will be applied this naming scheme, and someone could come along and add a storage type other than SoftLayer and be surprised by the mutation of the storage name...

Let me know if this makes sense.